### DISCUSSION TITLE
EMoji's auto-completion plugin implementation has performance issues

### DISCUSSION THREAD

**Discussion by Alecyrus** [ueberdosis/tiptap#2209] [Feature Requests]
Tippy instances are not reused. Every time the popup display is triggered, the tippy instance is recreated, which results in a performance penalty of over 120ms.
![image](https://user-images.githubusercontent.com/13865581/143933510-794ef476-af6b-4a36-a1ad-cbfa60bf1787.png)

My idea is to deprecate Tippy and use a separate UI component that uses Transform animations to display the UI component with the positioning parameters returned by the Suggestion plugin.

While this is not an out-of-the-box Emoji plugin, it reuses the popup instance and reduces the overhead of creating and destroying tippy instances.

I'll use [prosemirror-autocomplete](https://curvenote.github.io/prosemirror-autocomplete/) later on instead of tiptap's Suggestion plugin.

**Comment by rowanc1** [comment]
That is the approach we have taken with `prosemirror-autocomplete`, let me know if there is anything that needs to be upgraded in that library, happy to make improvements there (I'm one of the maintainers).

**Comment by philippkuehn** [comment]
You see this lag in the [official emoji demo](https://tiptap.dev/api/nodes/emoji#usage)?

**Comment by Alecyrus** [comment]
I am very sorry, the root cause is that I have set some plugins to tippy by default. These plugins will modify the class of the reference element when onShow, which causes the reflow.

This is my mistake, sorry for my unexamined judgment. @philippkuehn  @rowanc1
